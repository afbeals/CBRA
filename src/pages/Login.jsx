// Extenal
import React, { useEffect, useMemo } from 'react';
import { Redirect } from 'react-router';
import PropTypes from 'prop-types';

// Local
import { withUser, userUtility } from '~Modules/user';
import appEnum from '~Util/enum';
import Button from '~Components/Button';

// Constants
const classname = 'login';
const { DEX } = appEnum.APP.ROUTES;

// Component
const Login = ({ loginUser, getUserInfo, getIsLoggingIn, loginCancel }) => {
  const hasAccess = useMemo(
    () => getUserInfo && userUtility.getUserFromLocalStorage(),
    [getUserInfo],
  );

  useEffect(
    () => () => {
      if (getIsLoggingIn) {
        loginCancel();
      }
    },
    [getIsLoggingIn, loginCancel],
  );
  if (hasAccess) {
    return <Redirect to={DEX} />;
  }
  return (
    <div data-testid="login" className={classname}>
      <div className={`${classname}__group`}>
        <h2 className={`${classname}__group__title`}>
          Pokedex{' '}
          <span className={`${classname}__group__title__version`}>1.0</span>
        </h2>
        <Button
          text="Access"
          disabled={getIsLoggingIn}
          onClick={loginUser}
          icon={getIsLoggingIn ? 'fas fa-fan fa-spin' : 'fas fa-fan'}
          className={`${classname}__group__button`}
        />
      </div>
    </div>
  );
};

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  loginCancel: PropTypes.func.isRequired,
  getIsLoggingIn: PropTypes.bool.isRequired,
  getUserInfo: PropTypes.object,
};

export default withUser(Login);

export { Login };
