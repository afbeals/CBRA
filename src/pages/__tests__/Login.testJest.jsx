// External
import React from 'react';
import { render, cleanup } from '@testing-library/react';

// Local
import { Login } from '../Login';

// tests
describe('Login', () => {
  afterEach(cleanup);
  it('should match Login to snapshot', () => {
    const fnClick = jest.fn();
    const { getByTestId } = render(
      <Login
        loginUser={fnClick}
        loginCancel={fnClick}
        getIsLoggingIn={false}
      />,
    );
    const login = getByTestId('login');
    expect(login).toMatchSnapshot();
  });

  it('should contain an active access button', () => {
    const fnClick = jest.fn();
    const { getByText } = render(
      <Login
        loginUser={fnClick}
        loginCancel={fnClick}
        getIsLoggingIn={false}
      />,
    );
    const accessButton = getByText('Access');
    expect(accessButton).toBeTruthy();
  });
});
