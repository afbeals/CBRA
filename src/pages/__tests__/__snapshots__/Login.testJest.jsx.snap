// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match Login to snapshot 1`] = `
<div
  class="login"
  data-testid="login"
>
  <div
    class="login__group"
  >
    <h2
      class="login__group__title"
    >
      Pokedex
       
      <span
        class="login__group__title__version"
      >
        1.0
      </span>
    </h2>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained login__group__button MuiButton-containedPrimary"
      data-testid="button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Access
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium"
        >
          <i
            class="fas fa-fan"
          />
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
